pkg/reconciler/managed/reconciler.go

```diff
@@ -690,9 +690,18 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		record.Event(managed, event.Normal(reasonCreated, "Successfully requested creation of external resource"))
 
 		if creation.ResourceLateInitialized {
+			e := resource.NewRunnerWithExponentialBackoff(16 * time.Second)
 			// We use APIUpdatingApplicator to enforce the update to go through
 			// by setting object's resourceVersion to the latest.
-			if err := resource.NewAPIUpdatingApplicator(r.client).Apply(ctx, managed); err != nil {
+			c := resource.NewAPIUpdatingApplicator(r.client)
+			if err := e.Run(ctx, func() error {
+				err := c.Apply(ctx, managed)
+				if err != nil {
+					log.Debug(errUpdateManaged, "error", err, "requeue-after", time.Now().Add(r.shortWait))
+					record.Event(managed, event.Warning(reasonCannotUpdateManaged, errors.Wrap(err, errUpdateManaged)))
+				}
```

Comment:
How does this debugging info correspond to the debugging info below at line 705? Do we need both?